app/models/link.rb

```diff
@@ -8,7 +8,7 @@ class Link < Datafile
   has_one :preview, foreign_key: "datafiles_id", dependent: :destroy
 
   before_save :set_dates
-  after_commit :generate_preview_async
```

Comment:
This is what was making the (skipped) test fail. We were trying to generate a preview after every DB commit, including when a link had been deleted. So when we are trying to generate a preview for a non-existing link, we get an `record not found` error.